src/Defaults/AppChooserButton.vala

```diff
@@ -0,0 +1,347 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ * SPDX-FileCopyrightText: 2011-2025 elementary, Inc. (https://elementary.io)
+ */
+
+public class Defaults.AppChooserButton : Granite.Bin {
+    public string content_type { get; construct; }
+
+    public AppChooserButton (string content_type) {
+        Object (
+            content_type: content_type
+        );
+    }
+
+    construct {
+        var apps_store = new ListStore (typeof (AppInfo));
+
+        load_apps (apps_store, content_type);
```

Comment:
TODO:

- [ ] Leave a comment why we proceed even if the method fails
- [ ] Leave a comment why we call this method here